src/prefect/tasks/fivetran/fivetran.py

```diff
@@ -0,0 +1,160 @@
+        will poll Fivetran for connector status, and will only complete when the sync has completed or
+        when it receives an error status code from an API call.
+
+        Args:
+            - api_key (str): `API key` per https://fivetran.com/account/settings; should be secret!
+            - api_secret (str): `API secret` per https://fivetran.com/account/settings; should be secret!
+            - connector_id (str, optional): if provided, will overwrite value provided at init.
+            - poll_status_every_n_seconds (int, optional): this task polls the Fivetran API for status,
+                if provided this value will override the default polling time of 15 seconds.
+
+        Returns:
+            - dict: connector_id (str) and succeeded_at (timestamp str)
+        """
+
+        def parse_timestamp(api_time: str):
+            """Returns either the pendulum-parsed actual timestamp or
+            a very out-of-date timestamp if not set
+            """
+            return (
+                pendulum.parse(api_time)
+                if api_time is not None
+                else pendulum.from_timestamp(-1)
+            )
+
+        if not connector_id:
+            raise ValueError("Value for parameter `connector_id` must be provided.")
+        if not api_key:
+            raise ValueError("Value for parameter `api_key` must be provided.")
+        if not api_secret:
+            raise ValueError("Value for parameter `api_secret` must be provided.")
+
+        URL_CONNECTOR: str = "https://api.fivetran.com/v1/connectors/{}".format(
+            connector_id
+        )
+
+        self.logger.info(
+            "Attempting start of Fivetran connector {}, sleep time set to {} seconds.".format(
+                connector_id, poll_status_every_n_seconds
+            )
+        )
+
+        # Automatically call `raise_for_status` on every request
+        session = requests.Session()
+        session.hooks = {"response": lambda r, *args, **kwargs: r.raise_for_status()}
+        # Make sure connector configuration has been completed successfully and is not broken.
+        resp = session.get(URL_CONNECTOR, auth=(api_key, api_secret))
+        connector_details = resp.json()["data"]
+        URL_LOGS = "https://fivetran.com/dashboard/connectors/{}/{}/logs".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        URL_SETUP = "https://fivetran.com/dashboard/connectors/{}/{}/setup".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        setup_state = connector_details["status"]["setup_state"]
+        if setup_state != "connected":
+            EXC_SETUP: str = (
+                'Fivetran connector "{}" not correctly configured, status: {}; '
+                + "please complete setup at {}"
+            )
+            raise Exception(EXC_SETUP.format(connector_id, setup_state, URL_SETUP))
```

Comment:
Because most exceptions inherit from the base `Exception` class, just to avoid a weird edge case we should probably choose a more specific exception here and down below - maybe a `ValueError` for now?

Reply:
Ha, I was reading through the [list of built-in exceptions](https://docs.python.org/3/library/exceptions.html) and nothing really seemed to fit. I'll look again, but yeah maybe just ValueError.

Reply:
Yea, maybe we could use a [Prefect FAIL signal](https://docs.prefect.io/api/latest/engine/signals.html#fail) which simply indicates failure without having to pick an underlying exception.

Happy to merge without that, or happy to make that change myself post-merge, whatever you'd like